rep-2006.rst

```diff
@@ -0,0 +1,72 @@
+REP: 2006
+Title: ROS 2 Vulnerability Disclosure Policy
+Author: ROS 2 Security Working Group, Chris Lalancette <clalancette@openrobotics.org>
+Status: Active
+Type: Informational
+Content-Type: text/x-rst
+Created: 12-May-2020
+Post-History:
+
+
+Abstract
+========
+
+This REP describes the Vulnerability Disclosure Policy related to the ROS 2 Common Packages as defined in REP 2005.
+
+
+Motivation
+==========
+
+The developers of the Robot Operating System (ROS) take security very seriously.
+As such, we would like to be informed when a security bug is found so that it can be fixed and disclosed as quickly as possible.
+The rest of the document outlines what is covered by this policy and how to report security vulnerabilities.
+
+Scope
+=====
+
+The Vulnerability Disclosure Program outlined here covers all code within the ROS 2 Common Packages as defined in REP 2005.
+
+The ROS Security Working Group may also be able to assist you with reporting vulnerabilities which are not in scope of this policy but which are related to ROS or more generally to robotics.
+An increasing number of companies are bringing ROS-based products to market, and during this growth period we anticipate vulnerabilities to be identified in products before vendors have an established vulnerability reporting program.
+We will work with you on a best-effort basis to help connect you with the responsible vendor.
```

Comment:
Understood--the reporter simply sees a vulnerability, but that vuln could have been introduced anywhere along the supply chain. Although the scenario is complex, this paragraph interpreted as "best effort" only, not a commitment to any specific actions or timeline, and the burden should fall mainly on individuals receiving inbound reports.

Although the policy could stand without this paragraph, it's included specifically for outreach. It seems we should help address the current dearth of robot vuln reporting (of which you're well aware!). If we're able to kickstart reporting to the point where this becomes burdensome, we can claim success and update the REP to provide hands-off redirection. To my knowledge the most relevant and active CERT would be [ICS-CERT](https://www.us-cert.gov/ics) although they seem primarily focused on embedded systems.

Good point on the suggested change--the vendor may not be the one to address. Does this help clarify further?
```suggestion
We will work with you on a best-effort basis to help connect you with responsible parties best suited to address your concerns.
```

Reply:
> Who's going to be in charge of maintaining a list of of affected vendors (per ROS software package I guess) and distributing the information coherently? Open Robotics? If so, this will require a considerable amount of resources.

No, Open Robotics is definitely not going to commit to this.

This section has seemed to cause a lot of confusion up until this point.  To me, it is fairly clear that this is a "we might be able to help you find the responsible parties, since we know the robotics community.  But we don't guarantee anything".  But others aren't reading it this way.  I guess we can go with @vmayoral's suggestion here, but I'm thinking we should just remove this paragraph completely.  @ros2/security_working_group Thoughts?

Reply:
Having the chance to derive the load to third parties is the way to go. Either @SidFaber's suggestion or the one I initially propose would do. ICS-CERT is definitely the most popular but we're pretty aligned with [INCIBE-CERT](https://www.incibe-cert.es/en) and have interacted with a few others in EU. There're [big lists of reponse teams out there](https://www.first.org/members/teams/#CERT) and we probably should make a list of which ones should be considered (though this is a Security WG conversation).